packages/sdk-js/src/Identity.ts

```diff
@@ -0,0 +1,456 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { u8aEq } from '@polkadot/util'
+import type { Keypair } from '@polkadot/util-crypto/types'
+
+import { Blockchain } from '@kiltprotocol/chain-helpers'
+import { ConfigService } from '@kiltprotocol/config'
+import type { Issuer } from '@kiltprotocol/credentials'
+import {
+  NewDidVerificationKey,
+  authorizeTx,
+  createLightDidDocument,
+  didKeyToVerificationMethod,
+  getFullDid,
+  getStoreTx,
+  multibaseKeyToDidKey,
+  resolve,
+} from '@kiltprotocol/did'
+import type {
+  Did,
+  DidDocument,
+  DidUrl,
+  KeyringPair,
+  KiltAddress,
+  SignerInterface,
+  UriFragment,
+} from '@kiltprotocol/types'
+import { Crypto, SDKErrors, Signers } from '@kiltprotocol/utils'
+
+/**
+ * An Identity represents a DID and signing keys associated with it.
+ */
+export interface Identity {
+  did: Did
+  didDocument: DidDocument
+  signers: SignerInterface[]
+  /**
+   * Adds one or more signer interfaces to the `signers`.
+   *
+   * @param signers Signer Interface(s).
+   * @returns The (in-place) modified Identity object for chaining.
+   */
+  addSigner: (...signers: SignerInterface[]) => Identity
+  /**
+   * Convenience function similar to {@link addSigner}, but creates signers for all matching known algorithms from a keypair.
+   *
+   * @param keypairs
+   * @returns The (in-place) modified Identity object for chaining.
+   */
+  addKeypair: (...keypairs: Array<Keypair | KeyringPair>) => Promise<Identity>
```

Comment:
this name may be problematic; if I implement a method that helps rotate or add public keys to the DID document, these namings may clash